crates/task-impls/src/consensus.rs

```diff
@@ -539,14 +535,49 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                     );
                     consensus.saved_leaves.insert(leaf.commit(), leaf.clone());
 
+                    // If we are missing the parent from storage, the safety check will fail.  But we can
+                    // still vote if the liveness check succeeds.
+                    let liveness_check = justify_qc.get_view_number() > consensus.locked_view;
+
+                    if justify_qc.get_view_number() > consensus.high_qc.view_number {
+                        consensus.high_qc = justify_qc;
+                    }
```

Comment:
Should we do this only if the liveness check passes?

Reply:
I was thinking about that too.  My thought was that we should do it even if the liveness check fails.  So long as the QC is valid, then it is a high QC (if this check passes).  We may not vote on this proposal if the liveness check fails, but I think we still want to store the high QC.

Reply:
Makes sense! Then I think we should also update the high QC in the case of having the parent leaf.

Reply:
Also I think if this qc has a higher view number than our high qc the liveness check must have passed because I don't think it's possible (or shouldn't be possible) for `locked_qc.view > high_qc.view`